pkg/hope/hypervisors/proxmox_hypervisor.go

```diff
@@ -0,0 +1,275 @@
+	pm := ParameterMap(node.Parameters)
+	if insecure, ok := pm["INSECURE"]; ok {
+		switch insecure {
+		case "true", "1":
+			p.insecure = true
+		case "false", "0":
+			p.insecure = false
+		default:
+			errs = append(errs, fmt.Errorf("unknown value '%s' for INSECURE in Proxmox hypervisor", insecure))
+		}
+		delete(pm, "INSECURE")
+	}
+
+	for key := range pm {
+		errs = append(errs, fmt.Errorf("unknown property '%s' in Proxmox hypervisor", key))
+	}
+
+	proxmoxToken := os.Getenv("PROXMOX_API_TOKEN")
+	p.pc = proxmox.NewApiClient(p.node.User, p.node.Host, proxmoxToken)
+
+	if p.insecure {
+		p.pc.Client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+	}
+
+	return errors.Join(errs...)
+}
+
+func (p *ProxmoxHypervisor) ListNodes() ([]string, error) {
+	return p.pc.GetVmNames(p.node.Name)
+}
+
+func (p *ProxmoxHypervisor) ListBuiltImages(vms hope.VMs) ([]string, error) {
+	// With the right adaptation of arguments, it might be possible to ask
+	// other ProxmoxHypervisors for their list of built images.
+	// With that, this client could return some union of all available
+	// templates, rather than just the node attached to this instance.
+	return p.pc.GetTemplateNames(p.node.Name)
+}
+
+func (p *ProxmoxHypervisor) ListAvailableImages(hope.VMs) ([]string, error) {
+	return p.pc.GetTemplateNames(p.node.Name)
+}
+
+func (p *ProxmoxHypervisor) ResolveNode(node hope.Node) (hope.Node, error) {
+	ip, err := p.VMIPAddress(node.Name)
+	if err != nil {
+		return hope.Node{}, err
+	}
+
+	node.Hypervisor = ""
+	node.Host = ip
+	return node, nil
+}
+
+func (p *ProxmoxHypervisor) UnderlyingNode() (hope.Node, error) {
+	return p.node, nil
+}
+
+func (p *ProxmoxHypervisor) CopyImageMode() CopyImageMode {
+	return CopyImageModeFromFirst
+}
+
+func (p *ProxmoxHypervisor) CopyImage(vms hope.VMs, vmImageSpec hope.VMImageSpec, originalHV Hypervisor) error {
+	originalProxmoxHypervisor, ok := originalHV.(*ProxmoxHypervisor)
+	if !ok {
+		originalNode, err := originalHV.UnderlyingNode()
+		if err != nil {
+			return fmt.Errorf("failed to get node from non-proxmox hypervisor in image copy")
+		}
+		return fmt.Errorf("failed to get node from non-proxmox hypervisor %s in image copy", originalNode.Name)
+	}
+
+	tempNodeName := uuid.New().String()
+	if err := p.pc.CreateNodeFromTemplate(originalProxmoxHypervisor.node.Name, tempNodeName, vmImageSpec.Name); err != nil {
+		return err
+	}
+
+	if err := originalProxmoxHypervisor.waitForNode(5*time.Second, 5*time.Minute, tempNodeName); err != nil {
+		return err
+	}
+
+	// Make sure the template is actually being treated like a template.
+	config := map[string]interface{}{}
+	config["template"] = 1
+	if err := originalProxmoxHypervisor.pc.ConfigureNode(originalProxmoxHypervisor.node.Name, tempNodeName, config); err != nil {
+		return err
+	}
+
+	if err := p.pc.MigrateTemplate(originalProxmoxHypervisor.node.Name, p.node.Name, tempNodeName, p.insecure); err != nil {
```

Comment:
This migration runs way too slowly. Applying `migration_type=insecure` to these requests fails using an API Token, even with admin permissions, so it's feeling like this may be best implemented using backups + restores, including adjusting the packer build process to scp the backup to hope's packer images dir, rather than relying on in-cluster migrations. 

That might be the fastest path to copy vms between nodes. 

There could be room to reconsider this path at some point, or maybe just to have 2 different proxmox hypervisor implementations, one for local-only storage, and one for shared, where migrations like this aren't needed at all.